kratos/tests/cpp_tests/sources/test_parameters.cpp

```diff
@@ -41,12 +41,12 @@ namespace Kratos {
             KRATOS_CHECK((*i_parameter).IsBool());
             KRATOS_CHECK_EQUAL((*i_parameter).GetBool(), true);
             ++i_parameter;
-            KRATOS_CHECK(i_parameter->IsDouble());
-            KRATOS_CHECK_EQUAL(i_parameter->GetDouble(), 2.0);
+            KRATOS_CHECK((*i_parameter).IsDouble());
```

Comment:
Is it not possible to just override the -> operator, this is a very bizarre way to use a iterator

Reply:
`operator->` needs to return a pointer to `Parameters`, I don't know of any way around that.

The problem is that `Parameters` is not a container and doesn't really store any items, thus cannot provide stable references/pointers to "its" items. Our two options are:
- create a much more complex `Parameters` and iterator adaptor class that is capable of providing stable references/pointers
- drop the requirement of providing references, and remove the few cases where `operator->` is used (this is what this PR does). It's not a great solution because the standard clearly requires that iterators return references when dereferenced, so the iterator adaptors should not be used in stl algorithms in this form. However, the same can be said of the iterator adaptors currently in master too, with the added drawback that the related bug cannot be caught at compile time. At least this PR fixes that.

Reply:
i am a bit lost, what is the bug we are fixing by this?

i should say upfront that the Parameters object is very weird, but we had as a design constraint to be able to replace in glight the underlying json implementation and i could not find  a better way of doing it

Reply:
> Is it not possible to just override the -> operator?

another option is to provide a pointer-like interface to `Parameters`.
```cpp
class Parameters
{
    ...
    
    Parameters& operator*() {return *this;}

    const Parameters& operator*() const {return *this;}

    Parameters* operator->() {return this;}

    const Parameters* operator->() const {return this;}

    ...
};
```
but this is too esoteric for my taste.

Reply:
Still not big fan of new interface, the behaviour it is not the standard of an iterator